src/main/java/com/example/weatherwaveapi/util/urlbuilder/OpenWeatherUrlBuilder.java

```diff
@@ -2,21 +2,20 @@
 
 import com.example.weatherwaveapi.config.GeneralSettings;
 import jakarta.ws.rs.core.UriBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 @Component
 public class OpenWeatherUrlBuilder {
-    private static final String API_KEY = "api.key";
-    private static final String WEATHER_URL = "api.weatherUrl";
-    private static final String FORECAST_URL = "api.forecastUrl";
     private final String apiKey;
     private final String weatherApiUrl;
     private final String forecastApiUrl;
 
+    @Autowired
```

Comment:
Please do not use @Autowired: 
1) add @RequiredArgsConstructor annotation to the service
2) delete private final String apiKey; private final String weatherApiUrl; private final String forecastApiUrl;
3) create a private final instance of GeneralSettings generalSettings
4) use generalSettings.getApi().getOpenWeather().getUrl and so on